contracts/src/ERC20Destination.sol

```diff
@@ -90,9 +88,15 @@ contract ERC20Destination is IERC20Bridge, TeleporterTokenDestination, ERC20 {
 
     /**
      * @dev See {TeleporterTokenDestination-_deposit}
+     *
+     * Note: The amount returned must be the amount credited as a result of the transfer.
+     * For a standard ERC20 implementation such as this contract, that is equal to the full amount given.
+     * For fee/burn on transfer tokens, that amount could be less.
      */
     function _deposit(uint256 amount) internal override returns (uint256) {
-        return SafeERC20TransferFrom.safeTransferFrom(this, amount);
```

Comment:
The use of the library here was previously making an external call to the same contract using `address(this)`, which costs more gas than necessary and should be avoided when only making calls internal to this contract itself.